pkg/platform/api/graphql/model/buildplanner/buildExpression.go

```diff
@@ -0,0 +1,245 @@
+package model
+
+import (
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/logging"
+)
+
+type Comparator string
+
+type Operation int
+
+const (
+	ComparatorEQ  Comparator = "eq"
+	ComparatorGT             = "gt"
+	ComparatorGTE            = "gte"
+	ComparatorLT             = "lt"
+	ComparatorLTE            = "lte"
+	ComparatorNE             = "ne"
+
+	OperationAdd Operation = iota
+	OperationRemove
+	OperationUpdate
+
+	SolveFuncName           = "solve"
+	SolveLegacyFuncName     = "solve_legacy"
```

Comment:
For the current build expressions the solve function is called `solve_legacy` from my understanding this will soon be updated to `solve`

Reply:
If this is meant to happen before we do the state tool release then I'd rather we not have this fallback. Ie. implement it the way we expect it to work when we go live. That would achieve several things: 

* We don't accidentally release with fallback code that can have bugs.
* We don't end up with throwaway code in our codebase.
* Perhaps most importantly: It puts pressure on extensibility to remove this blocker rather than allow it to slip into a release.

Reply:
After discussions with PB, this isn't meant to happen before we do a release of the state tool. In fact, they could potentially be adding a new `camel` application that we would also have to search for alongside `solve` and `solve_legacy`. I've filed a follow-up ticket for the `camel` application and this code will have to stay in place for the foreseeable future.